src/main/kotlin/app/Main.kt

```diff
@@ -3,60 +3,112 @@
-import app.utils.CommandExplore
+import app.utils.CommandAdd
+import app.utils.CommandList
+import app.utils.CommandRemove
 import app.utils.Options
+import app.utils.RepoHelper
 import com.beust.jcommander.JCommander
-import com.beust.jcommander.ParameterException
-import kotlin.system.exitProcess
 
 fun main(argv: Array<String>) {
-    val commandExplore = "explore"
+    val commandAdd = "add"
     val commandConfig = "config"
+    val commandList = "list"
+    val commandRemove = "remove"
 
     val options = Options()
-    val explore = CommandExplore()
+    val add = CommandAdd()
     val config = CommandConfig()
+    val list = CommandList()
+    val remove = CommandRemove()
     val jc: JCommander = JCommander.newBuilder()
             .programName("sourcerer")  // Used for usage method.
             .addObject(options)
-            .addCommand(commandExplore, explore)
+            .addCommand(commandAdd, add)
             .addCommand(commandConfig, config)
+            .addCommand(commandList, list)
+            .addCommand(commandRemove, remove)
             .build()
 
-    Configurator.setCurrentOptions(options)
+    jc.parse(*argv)
+    Configurator.options = options
 
-    try {
-        jc.parse(*argv)
-    } catch (e: ParameterException) {
-        println(e.message)
-        exitProcess(1)  // Reporting failure with non-zero exit code.
+    if (options.help) {
+        help(jc)
+        return
+    }
+
+    if (options.setup) {
+        setup()
+        return
     }
 
     when (jc.parsedCommand) {
-        commandExplore -> explore(explore)
+        commandAdd -> add(add)
         commandConfig -> config(config)
-        else -> help(jc)  // Show help info if no command specified.
+        commandList -> list(list)
+        commandRemove -> remove(remove)
+        else -> ui()
     }
-
-    println(arrayOf(options.username, options.password).joinToString())
-    println(jc.parsedCommand)
-    println(explore.path)
 }
 
-fun explore(commandOptions: CommandExplore) {
-    RepoExplorer(commandOptions).explore()
+fun add(commandAdd: CommandAdd) {
+    val path = commandAdd.path
+    if (path != null && RepoHelper.isValidRepo(path)) {
+        Configurator.user.addRepo(path)
+        Configurator.saveToFile()
+        println("Added git repository at $path.")
+    } else {
+        println("No valid git repository found at $path.")
+    }
 }
 
 fun config(commandOptions: CommandConfig) {
-    val options = Configurator.createOptions(commandOptions.pair)
-    val userDir = Configurator.userDir
-    if (userDir != null) {
-        Configurator.saveConfig(userDir, options)
+    Configurator.setConfig(commandOptions.pair[0], commandOptions.pair[1])
+}
+
+fun list(commandList: CommandList) {
+    if (Configurator.config.repos.isNotEmpty()) {
+        println("Tracked repositories:")
+        var counter = 0
+        for (repo in Configurator.config.repos) {
+            counter++
+            println("$counter. " + repo.path)
```

Comment:
You don't really need numbers for them unless they can be used for some purpose.